singer_encodings/jsonl.py

```diff
@@ -0,0 +1,47 @@
+import itertools
+import json
+
+import singer
+
+LOGGER = singer.get_logger()
+
+def get_JSONL_iterators(iterator, options):
+    """Function to return data iterator for JSONL files"""
+    # Get JSOL rows
+    records = get_JSONL_rows(iterator)
+    check_jsonl_sample_records, records = itertools.tee(records)
+
+    # Veirfy the 'date_overrides' and 'key_properties' as per the config
+    check_key_properties_and_date_overrides_for_jsonl_file(options, check_jsonl_sample_records)
+    return records
+
+def check_key_properties_and_date_overrides_for_jsonl_file(options, jsonl_sample_records):
+    """
+    Function to check the 'key_properties' and 'date_overrides'
+    are provided in the JSON Data as expected from the config
+    """
+    for record in jsonl_sample_records:
+        keys = record.keys()
+
+        if options.get('key_properties'):
```

Comment:
Please explain if conditions.

Reply:
Added comments for validation of key properties and date overrides.